client/src/Components/Settings/passwordModal.jsx

```diff
@@ -1,33 +1,76 @@
 import React, { Component } from 'react'
-import { Button, Modal, Form } from 'semantic-ui-react'
+import { Button, Modal, Form, Message, Divider } from 'semantic-ui-react'
 
 export default class PasswordModal extends Component {
     state = { 
         open: false,
         oldPassword: '',
         newPassword: '',
         checkPassword: '',
+        disabled: true,
+        message: ''
      }
 
     close = () => this.setState({ open: false })
 
     show = () => this.setState({ open: true })
 
     handleChange = (e) => {
+        const { newPassword, oldPassword } = this.state;
```

Comment:
Could you help me understand this logic?

Reply:
This handle change is checking the input field that the user is typing in first and will handle logic depending on if they're typing in the new password for the first time or the second one that checks that it matches. If the first one, it sets to state right away; but if the second, it checks that they match before setting to state. Disabled and message are just adjusting what renders for the user's feedback and only allows the Update button to become active when all fields are properly filled.